### PR TITLE
Enhancement/azure storage 3.8

### PR DESCRIPTION
**Pull request by SophieDorval**
# Description
Add Rule 3.8 from Azure CISv1.2.0 Benchmark: 3.8 Ensure soft delete is enabled for Azure Storage (Automated)

**Make sure the PR is against the `develop` branch (see [Contributing](https://github.com/nccgroup/ScoutSuite/blob/master/CONTRIBUTING.md)).**

Please include a summary of the change(s) and which issue(s) it addresses. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Select the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/output/data/html/partials/azure/services.storageaccounts.subscriptions.id.storage_accounts.html [hunk 1]
```diff
@@ -19,7 +19,12 @@ <h4 class="list-group-item-heading">Information</h4>
                 {{/if}}
             </span>
         </div>
-        <div class="list-group-item-text item-margin">Tags: 
+
+        {{#each blob_services}}
+            <div class="list-group-item-text item-margin">Storage Soft Delete: <span id="storageaccounts.subscriptions.{{@../key}}.storage_accounts.{{@key}}.blob_services.{{@key}}.soft_delete_enabled">{{convert_bool_to_enabled soft_delete_enabled}}</span></div>
```
Comment (j4v) [inline ScoutSuite/output/data/html/partials/azure/services.storageaccounts.subscriptions.id.storage_accounts.html:24]: The `id_suffix` is set but this is not being highlighted for some reason.

File: ScoutSuite/providers/azure/rules/findings/storageaccount-soft-delete-enabled.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+    "description": "Storage soft delete is enabled",
```
Comment (j4v) [inline ScoutSuite/providers/azure/rules/findings/storageaccount-soft-delete-enabled.json:2]: All words start with an uppercase - you can just run https://github.com/nccgroup/ScoutSuite/tree/master/tools#format_findingspy to take care of formatting.

File: requirements.txt [hunk 3]
```diff
@@ -31,13 +31,13 @@ httplib2shim>=0.0.3
 
 # Azure Provider
 ## core requirements
-azure-cli-core==2.12.0
+azure-cli-core==2.18.0
```
Comment (j4v) [inline requirements.txt:34]: By updating these dependencies, some authentication mechanisms no longer work.

For `--user-account` authentication I'm getting:
```
2021-02-05 11:47:11 wrkbx2 scout[319] ERROR storageaccounts.py L27: Failed to retrieve storage accounts: 'UserPassCredentials' object has no attribute 'get_token'
```

**Review by j4v — CHANGES_REQUESTED** [review]

**Comment by j4v** [comment]
Looks mostly good, but by upgrading the dependencies some authentication mechanisms will no longer work. That's expected though, as per https://github.com/nccgroup/ScoutSuite/issues/862.

File: ScoutSuite/output/data/html/partials/azure/services.storageaccounts.subscriptions.id.storage_accounts.html (hunk 1 above)
Comment (j4v) [reply ScoutSuite/output/data/html/partials/azure/services.storageaccounts.subscriptions.id.storage_accounts.html]: This is now fixed.

File: ScoutSuite/providers/azure/rules/findings/storageaccount-soft-delete-enabled.json (hunk 2 above)
Comment (SophieDorval) [reply ScoutSuite/providers/azure/rules/findings/storageaccount-soft-delete-enabled.json]: Done

File: requirements.txt (hunk 3 above)
Comment (SophieDorval) [reply requirements.txt:34]: Done in another PR

**Review by j4v — APPROVED** [review]

**Pull request merged**